File: backend/app/core/task_state.py
from typing import Final

STATUS_QUEUED: Final[str] = "queued"
STATUS_RUNNING: Final[str] = "running"
STATUS_SUCCEEDED: Final[str] = "succeeded"
STATUS_FAILED: Final[str] = "failed"
STATUS_CANCELLED: Final[str] = "cancelled"

TERMINAL_STATUSES: Final[set[str]] = {
    STATUS_SUCCEEDED,
    STATUS_FAILED,
    STATUS_CANCELLED,
}

CANCELLABLE_STATUSES: Final[set[str]] = {
    STATUS_QUEUED,
    STATUS_RUNNING,
}

RETRYABLE_STATUSES: Final[set[str]] = {
    STATUS_FAILED,
    STATUS_CANCELLED,
}

REPORT_READY_STATUSES: Final[set[str]] = {
    STATUS_SUCCEEDED,
}

REPORT_ACCESSIBLE_STATUSES: Final[set[str]] = {
    STATUS_SUCCEEDED,
    STATUS_FAILED,
}


def is_terminal(status: str) -> bool:
    return status in TERMINAL_STATUSES


def is_cancellable(status: str) -> bool:
    return status in CANCELLABLE_STATUSES


def is_retryable(status: str) -> bool:
    return status in RETRYABLE_STATUSES


def can_fetch_report(status: str) -> bool:
    return status in REPORT_ACCESSIBLE_STATUSES

File: frontend/src/constants/taskState.js
export const TASK_STATUS = {
  QUEUED: "queued",
  RUNNING: "running",
  SUCCEEDED: "succeeded",
  FAILED: "failed",
  CANCELLED: "cancelled",
};

export const TERMINAL_STATUSES = new Set([
  TASK_STATUS.SUCCEEDED,
  TASK_STATUS.FAILED,
  TASK_STATUS.CANCELLED,
]);

export const CANCELLABLE_STATUSES = new Set([
  TASK_STATUS.QUEUED,
  TASK_STATUS.RUNNING,
]);

export const RETRYABLE_STATUSES = new Set([
  TASK_STATUS.FAILED,
  TASK_STATUS.CANCELLED,
]);

export const REPORT_VIEWABLE_STATUSES = new Set([
  TASK_STATUS.SUCCEEDED,
  TASK_STATUS.FAILED,
]);

export const STATUS_LABEL = {
  [TASK_STATUS.SUCCEEDED]: "成功",
  [TASK_STATUS.FAILED]: "失败",
  [TASK_STATUS.RUNNING]: "运行中",
  [TASK_STATUS.QUEUED]: "排队中",
  [TASK_STATUS.CANCELLED]: "已取消",
};

export function getStatusLabel(status) {
  return STATUS_LABEL[status] || status || "未知";
}
